lib/streamlit/elements/camera_image_input.py

```diff
@@ -0,0 +1,156 @@
+
+        camera_image_input_proto = CameraImageInputProto()
+        camera_image_input_proto.form_id = current_form_id(self.dg)
+
+        if help is not None:
+            camera_image_input_proto.help = dedent(help)
+
+        def serialize_camera_image_input(
+            snapshot: SomeUploadedSnapshotFile,
+        ) -> FileUploaderStateProto:
+            state_proto = FileUploaderStateProto()
+
+            ctx = get_report_ctx()
+            if ctx is None:
+                return state_proto
+
+            # ctx.uploaded_file_mgr._file_id_counter stores the id to use for
+            # the *next* uploaded file, so the current highest file id is the
+            # counter minus 1.
+            state_proto.max_file_id = ctx.uploaded_file_mgr._file_id_counter - 1
+
+            if not snapshot:
+                return state_proto
+
+            file_info: UploadedFileInfoProto = state_proto.uploaded_file_info.add()
+            file_info.id = snapshot.id
+            file_info.name = snapshot.name
+            file_info.size = snapshot.size
+
+            return state_proto
+
+        def deserialize_camera_image_input(
+            ui_value: Optional[FileUploaderStateProto], widget_id: str
+        ) -> SomeUploadedSnapshotFile:
+            file_recs = self._get_file_recs(widget_id, ui_value)
+
+            if len(file_recs) == 0:
+                return_value = None
+            else:
+                return_value = UploadedFile(file_recs[0])
+            return return_value
+
+        widget_value, _ = register_widget(
+            "camera_image_input",
+            camera_image_input_proto,
+            user_key=key,
+            on_change_handler=on_change,
+            args=args,
+            kwargs=kwargs,
+            deserializer=deserialize_camera_image_input,
+            serializer=serialize_camera_image_input,
+        )
+
+        ctx = get_report_ctx()
+        camera_image_input_state = serialize_camera_image_input(widget_value)
+
+        uploaded_shapshot_info = camera_image_input_state.uploaded_file_info
+
+        if ctx is not None and len(uploaded_shapshot_info) != 0:
+            newest_file_id = camera_image_input_state.max_file_id
+            active_file_ids = [f.id for f in uploaded_shapshot_info]
+
+            ctx.uploaded_file_mgr.remove_orphaned_files(
+                session_id=ctx.session_id,
+                widget_id=camera_image_input_proto.id,
+                newest_file_id=newest_file_id,
+                active_file_ids=active_file_ids,
+            )
+
+        self.dg._enqueue("camera_image_input", camera_image_input_proto)
+        return cast(SomeUploadedSnapshotFile, widget_value)
+
+    @property
+    def dg(self) -> "streamlit.delta_generator.DeltaGenerator":
+        """Get our DeltaGenerator."""
+        return cast("streamlit.delta_generator.DeltaGenerator", self)
+
+    @staticmethod
+    def _get_file_recs(
```

Comment:
I'm confused as to what this method is supposed to do. It looks like we are using file_uploader logic to grab the files based on the widget idea but why is it returning a list? Shouldn't we just have the method check that it's a length of one and then return that file if it is a length of one? It looks like the deserializing method does something similar

What does the `recs` in  `get_file_recs` mean?